homeassistant/components/shelly/entity.py

```diff
@@ -395,9 +398,9 @@ def __init__(
             self._name = get_entity_name(
                 self.wrapper.device, block, self.description.name
             )
-        else:
+        elif entry is not None:
             self._unique_id = entry.unique_id
-            self._name = entry.original_name
+            self._name = str(entry.original_name)
```

Comment:
Why wrap in String, what can it be? If it can be None, we should allow it to be `None`